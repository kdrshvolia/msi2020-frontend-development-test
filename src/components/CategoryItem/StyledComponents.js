import styled, { css } from 'styled-components';

const Item = styled.div`
  background: ${(props) => (props.isChosen ? '#f8f8f8' : '#FFFFFF')};
  color: ${(props) => (props.isChosen ? '#333333' : '#ababab')};
  display: flex;
  height: 28px;
  cursor: pointer;
  border: 2px solid #f8f8f8;
  box-sizing: border-box;
  border-radius: 6px;
  padding: 6px 15px;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  letter-spacing: 2px;
  transition: 0.7s;
  margin: 5px 0 0;
`;

export { Item };
