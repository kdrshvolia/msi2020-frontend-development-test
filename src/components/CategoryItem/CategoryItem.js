import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { Item } from './StyledComponents';

const CategoryItem = ({ id, onClick, children, isChosen }) => {
  const handleClick = useCallback(() => {
    onClick(id);
  }, [id]);

  return (
    <Item isChosen={isChosen} onClick={handleClick}>
      {children}
    </Item>
  );
};

CategoryItem.propTypes = {
  id: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};

export default CategoryItem;
