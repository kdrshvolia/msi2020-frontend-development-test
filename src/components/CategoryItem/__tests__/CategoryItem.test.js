import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import CategoryItem from '../CategoryItem';

describe('CategoryItem', () => {
  it('setItem on click', () => {
    const isClicked = jest.fn();
    const id = '1';
    const wrapper = mount(<CategoryItem onClick={isClicked} id={id} />);
    const item = wrapper.find('div');
    item.simulate('click');
    expect(isClicked).toBeCalledWith(id);
  });

  it('renders correctly', () => {
    const id = '1';
    const isClicked = jest.fn();
    const children = 'text';
    const rendered = renderer.create(<CategoryItem onClick={isClicked} id={id} isChosen={false}>{children}</CategoryItem>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
