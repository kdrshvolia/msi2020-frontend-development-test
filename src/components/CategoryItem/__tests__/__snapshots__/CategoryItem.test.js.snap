// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem renders correctly 1`] = `
<div
  className="sc-AxjAm bctaSZ"
  onClick={[Function]}
>
  text
</div>
`;
